lib/cppcheck.cpp

```diff
@@ -618,26 +686,65 @@ void CppCheck::executeRules(const std::string &tokenlist, const Tokenizer &token
         if (rule.pattern.empty() || rule.id.empty() || rule.severity == Severity::none || rule.tokenlist != tokenlist)
             continue;
 
-        const char *error = nullptr;
+        const char *pcreCompileErrorStr = nullptr;
         int erroffset = 0;
-        pcre *re = pcre_compile(rule.pattern.c_str(),0,&error,&erroffset,nullptr);
+        pcre * const re = pcre_compile(rule.pattern.c_str(),0,&pcreCompileErrorStr,&erroffset,nullptr);
         if (!re) {
-            if (error) {
-                ErrorLogger::ErrorMessage errmsg(std::list<ErrorLogger::ErrorMessage::FileLocation>(),
-                                                 emptyString,
-                                                 Severity::error,
-                                                 error,
-                                                 "pcre_compile",
-                                                 false);
+            if (pcreCompileErrorStr) {
+                const ErrorLogger::ErrorMessage errmsg(std::list<ErrorLogger::ErrorMessage::FileLocation>(),
+                                                       emptyString,
+                                                       Severity::error,
+                                                       pcreCompileErrorStr,
+                                                       "pcre_compile",
+                                                       false);
 
                 reportErr(errmsg);
             }
             continue;
         }
 
+        // Optimize the regex, but only if PCRE_CONFIG_JIT is available
+#ifdef PCRE_CONFIG_JIT
+        const char *pcreStudyErrorStr = nullptr;
```

Comment:
you could reuse `error`. it is null here. well.. it does not hurt to have two separate pointers so do as you think is best.

Reply:
I left it as it is. IMHO, having separate pointers is good, which potentially simplifies later refactoring.